File: 20221005/모범답안/20221005_01/CountTruncateLetter_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.CountTruncateLetter.CountTruncateLetter import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'Hello world Nice world'

        # when
        result = solution(s)

        # then
        self.assertEqual(3, result)

    def test_2(self):
        # given
        s = 'Hello world Nice World'

        # when
        result = solution(s)

        # then
        self.assertEqual(4, result)

    def test_3(self):
        # given
        s = 'Apple naver Kakao Naver'

        # when
        result = solution(s)

        # then
        self.assertEqual(4, result)

    def test_4(self):
        # given
        s = 'Where there is a will there is a way'

        # when
        result = solution(s)

        # then
        self.assertEqual(6, result)

    def test_5(self):
        # given
        s = 'The road to success and the road to failure are almost exactly the same'

        # when
        result = solution(s)

        # then
        self.assertEqual(11, result)

    def test_6(self):
        # given
        s = 'It is better to fail in originality than to succeed in imitation'

        # when
        result = solution(s)

        # then
        self.assertEqual(10, result)


if __name__ == '__main__':
    unittest.main()

File: 20221005/모범답안/20221005_01/CountTruncateLetter_test.js
import assert from 'assert'
import solution from './CountTruncateLetter.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'Hello world Nice world'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'Hello world Nice World'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'Apple naver Kakao Naver'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'Where there is a will there is a way'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(6, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'The road to success and the road to failure are almost exactly the same'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(11, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'It is better to fail in originality than to succeed in imitation'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(10, result)
    done();
  });
})
